modules/generative-openai/clients/openai.go

```diff
@@ -0,0 +1,189 @@
+	prompt := v.generatePrompt(text, task, language)
+
+	settings := config.NewClassSettings(cfg)
+
+	body, err := json.Marshal(generateInput{
+		Prompt:           prompt,
+		Model:            settings.Model(),
+		MaxTokens:        settings.MaxTokens(),
+		Temperature:      settings.Temperature(),
+		FrequencyPenalty: settings.FrequencyPenalty(),
+		PresencePenalty:  settings.PresencePenalty(),
+		TopP:             settings.TopP(),
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "marshal body")
+	}
+
+	oaiUrl, err := url.JoinPath(v.host, v.path)
+	if err != nil {
+		return nil, errors.Wrap(err, "join OpenAI API host and path")
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", oaiUrl,
+		bytes.NewReader(body))
+	if err != nil {
+		return nil, errors.Wrap(err, "create POST request")
+	}
+	apiKey, err := v.getApiKey(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "OpenAI API Key")
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", apiKey))
+	req.Header.Add("Content-Type", "application/json")
+
+	res, err := v.httpClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "send POST request")
+	}
+	defer res.Body.Close()
+
+	bodyBytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "read response body")
+	}
+
+	var resBody generateResponse
+	if err := json.Unmarshal(bodyBytes, &resBody); err != nil {
+		return nil, errors.Wrap(err, "unmarshal response body")
+	}
+
+	if res.StatusCode > 399 {
+		if resBody.Error != nil {
+			return nil, errors.Errorf("failed with status: %d error: %v", res.StatusCode, resBody.Error.Message)
+		}
+		return nil, errors.Errorf("failed with status: %d", res.StatusCode)
+	}
+	textResponse := resBody.Choices[0].Text
+	if len(resBody.Choices) > 0 && textResponse != "" {
+		//nolint:gofumpt    //todo [@marcin, should we do this? seems like OpenAI returns the \n from the prompt I think for some reason]
+		replaceAll := strings.ReplaceAll(textResponse, "\n", "")
+		return &ent.GenerateResult{
+			Result: &replaceAll,
+		}, nil
+	}
+	return &ent.GenerateResult{
+		Result: nil,
+	}, nil
+}
+
+func (v *openai) generatePrompt(text string, question string, language string) string {
+	//nolint:gofumpt    //todo [byron - this is the prompt created by Bob/Connor, check if it actually performs better than simple piping
+	//todo of question + language check out code commented out below code]
+	return fmt.Sprintf(`We need your help to complete the task: %v
+	
+	Additional instructions are as follows:
+	- You must base your response on the result in JSON format and nothing else.
+	- The response should be in %v.
+	- You should only base your results on the result in JSON format. If you can't do this, could you explain why you can't do this?
+	- You should only respond with the result to the task, nothing more.
+	- You should be kind and decent.
+	
+	This is the result in JSON format:
+	{
```

Comment:
@byronvoorbach and @antas-marcin doesn't this result in wrong JSON because it is encapsulated in brackets?